components/brave_rewards/core/rewards_engine_helper.h

```diff
@@ -40,6 +43,16 @@ class RewardsEngineHelper {
   const raw_ref<RewardsEngineImpl> engine_;
 };
 
+// A mixin for exposing a user data key for a `RewardsEngineHelper` class.
+template <typename T>
+class WithHelperKey {
```

Comment:
What problem are you trying to solve here? I think this is making the rewards code harder for others to understand. If there's a generalized problem you think we need a solution for we should discuss and possibly submit to upstream.